kong/runloop/handler.lua

```diff
@@ -1389,10 +1389,8 @@ return {
       -- We overcome this behavior with our own logic, to preserve user
       -- desired semantics.
       -- perf: branch usually not taken, don't cache var outside
-      if byte(ctx.request_uri or var.request_uri, -1) == QUESTION_MARK then
-        var.upstream_uri = var.upstream_uri .. "?"
-      elseif var.is_args == "?" then
-        var.upstream_uri = var.upstream_uri .. "?" .. (var.args or "")
+      if var.is_args == "?" or byte(ctx.request_uri, -1) == QUESTION_MARK then
```

Comment:
I do not know why we delete `var.request_uri`?

Reply:
I thought since both values are equal due to this line-
```
      ctx.request_uri = var.request_uri
```
in the before function. I'll change it back to what it was.